src/sentry/integrations/vsts/vsts_subscription_script.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import absolute_import
+
+import logging
+
+from django.core.urlresolvers import reverse
+from uuid import uuid4
+
+from sentry.constants import ObjectStatus
+from sentry.integrations.exceptions import ApiError, ApiUnauthorized
+from sentry.models import OrganizationIntegration
+from sentry.utils.http import absolute_uri
+
+VSTS_WEBHOOK_URL = absolute_uri(reverse('sentry-extensions-vsts-issue-updated'))
+
+logger = logging.getLogger('sentry.integrations.vsts_script')
+
+
+def create_webhook(integration, organization_id):
+    installation = integration.get_installation(organization_id)
+    client = installation.get_client()
+    shared_secret = uuid4.hex + uuid4.hex
+
+    resp = client.create_subscription(
+        instance=installation.instance,
+        shared_secret=shared_secret,
+    )
+
+    installation.model.metadata['subscription'] = {
+        'id': resp['id'],
+        'secret': shared_secret,
+    }
+    installation.model.save()
+
+
+def recreate_subscriptions(self):
+    org_integrations = OrganizationIntegration.objects.filter(
+        integration__provider='vsts',
+        integration__status=ObjectStatus.VISIBLE,
+        status=ObjectStatus.VISIBLE,
+    ).select_related('integration')
+
+    # TODO(lb): this looks a little weird to me...
+    # can I run into issues with both values_list and distinct?
+    integration_ids = list(
```

Comment:
This is fine, but I'd convert this into a `set()` instead of a list. Since they are unique anyways, and later, you check `if integration not in integration_ids` also within a loop.